app/models/grda_warehouse/cas_project_client_calculator/tc_hmis_hat.rb

```diff
@@ -0,0 +1,315 @@
+        currently_fleeing: :hat_e13_ipv_fleeing,
+        dv_date: :hat_e14_ipv_date,
+        required_minimum_occupancy: :hat_a8_household_size,
+        # Internal
+        single_parent_child_over_ten: :hat_a7_single_parent,
+        legal_custody: :hat_a9_custody,
+        future_custody: :hat_a10_future_custody,
+        household_size: :hat_a8_household_size,
+      }.freeze
+    end
+    memoize :assessment_keys
+
+    private def boolean_lookups
+      [
+        :lifetime_sex_offender,
+        :foster_care,
+        :open_case,
+        :drug_test,
+        :heavy_drug_use,
+        :sober,
+        :willing_case_management,
+        :employed_three_months,
+        :living_wage,
+        :need_daily_assistance,
+        :full_time_employed,
+        :can_work_full_time,
+        :willing_to_work_full_time,
+        :rrh_successful_exit,
+        :th_desired,
+        :site_case_management_required,
+        :ongoing_case_management_required,
+        :currently_fleeing,
+      ].freeze
+    end
+    memoize :boolean_lookups
+
+    private def tc_hat_questions
+      [
+        :family_member,
+        :lifetime_sex_offender,
+        :strengths,
+        :challenges,
+        :foster_care,
+        :open_case,
+        :housing_for_formerly_homeless,
+        :neighborhood_ids_for_cas,
+        :cas_assessment_collected_at, # note this is really just assessment_collected_at
+        :days_homeless_in_last_three_years_cached,
+        :literally_homeless_last_three_years_cached,
+        :drug_test,
+        :heavy_drug_use,
+        :sober,
+        :willing_case_management,
+        :employed_three_months,
+        :living_wage,
+        :need_daily_assistance,
+        :full_time_employed,
+        :can_work_full_time,
+        :willing_to_work_full_time,
+        :rrh_successful_exit,
+        :lifetime_sex_offender,
+        :th_desired,
+        :drug_test,
+        :employed_three_months,
+        :site_case_management_required,
+        :ongoing_case_management_required,
+        :currently_fleeing,
+        :dv_date,
+        :va_eligible,
+        :vash_eligible,
+        :rrh_desired,
+        :required_minimum_occupancy,
+        :required_number_of_bedrooms,
+      ]
+    end
+
+    def most_recent_assessment_for_destination(client)
+      return unless client.present?
+
+      cas_assessment_collected_at(client)&.to_date&.to_s
+    end
+
+    private def cas_assessment(client)
+      client.source_clients.map do |source_client|
+        source_client.assessments.housing_needs.order(assessment_date: :desc).first
+      end.
+        compact.
+        max_by(&:assessment_date)
+    end
+
+    private def for_string(client, key)
+      cas_assessment(client).assessment_questions.find_by(assessment_question: assessment_keys[key])&.assessment_answer
+    end
+
+    private def for_boolean(client, key)
+      for_string(client, key)&.downcase == 'yes'
+    end
```

Comment:
Does it makes sense to move these `for_boolean` and `for_string` methods up to `AssessmentQuestion`?